activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+*   `AR::Relation#reverse_order` throws `AR::IrreversibleOrderError` when the order can not be reversed
```

Comment:
I think we should spell it out. I don't think we should make the assumption it's obvious for developers who are new to Rails.